lib/Ast_to_Mir.ml

```diff
@@ -64,11 +59,21 @@ let truncate_dist ast_obs t =
          (BinOp (obs, cond, trans_expr x), bind_loc x.expr_typed_loc add_inf, y))
   in
   match t with
-  | Ast.NoTruncate -> None
-  | Ast.TruncateUpFrom lb -> Some (trunc Less lb None)
-  | Ast.TruncateDownFrom ub -> Some (trunc Greater ub None)
+  | Ast.NoTruncate -> []
+  | Ast.TruncateUpFrom lb -> [trunc Less lb None]
+  | Ast.TruncateDownFrom ub -> [trunc Greater ub None]
   | Ast.TruncateBetween (lb, ub) ->
-      Some (trunc Less lb (Some (trunc Greater ub None)))
+      [trunc Less lb (Some (trunc Greater ub None))]
```

Comment:
This makes sense to me. Have you checked that it is in fact what the current Stan compiler does?

Reply:
I forget, maybe